File: C#JAVA_Client_Server/ClientServcerC#Protocols/Solution1/Repository/EchipaDBRepository.cs
using System;
using System.Collections.Generic;

using Domain;
using Domain.validators.interfaceAndException;
using MySql.Data.MySqlClient;

namespace Repository
{
    public class EchipaDBRepository: IObjectRepo<Echipa,int>
    {
        private IValidator<Echipa> _validator;
        private MySqlDataAdapter dataAdapter=new MySqlDataAdapter();
        private Utils.Utils _utils;

        public EchipaDBRepository(IValidator<Echipa> validator)
        {
            _validator = validator;
            _utils = new Utils.Utils();
        }

        public Echipa FindOne(int id)
        {
            var connection = _utils.GetConnection();
            var commander =_utils.GetCommander(connection);
            try
            {
                connection.Open();
                var sql = "select * from echipa where idEchipa=@idEchipa";
                commander = new MySqlCommand(sql, connection);
                commander.Parameters.AddWithValue("@idEchipa", id);
                MySqlDataReader reader = commander.ExecuteReader();
                Echipa echipa = new Echipa();
                reader.Read();
                echipa.NumeEchipa = reader.GetString(1);
                echipa.Id = reader.GetInt32(0);
                return echipa;
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
                throw;
            }
        }

        public IEnumerable<Echipa> FindAll()
        {
            var connection = _utils.GetConnection();
            var commander = _utils.GetCommander(connection);
            try
            {
                List<Echipa> echipe = new List<Echipa>();
                connection.Open();
                var sql = "select * from echipa";
                commander = new MySqlCommand(sql, connection);

                MySqlDataReader reader = commander.ExecuteReader();
                while (reader.Read())
                {
                    Echipa echipa = new Echipa(reader.GetString(1));
                    echipa.Id = reader.GetInt32(0);
                    echipe.Add(echipa);
                }

                return echipe;
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
                throw;
            }
        }

        public void Save(Echipa entity)
        {
            try
            {
                _validator.validate(entity);
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
                return;
            }
            var connection = _utils.GetConnection();
            var commander = _utils.GetCommander(connection);
            try
            {
                connection.Open();
                var sql = "Insert into echipa(numeEchipa) values(@numeEchipa)";
                commander = new MySqlCommand(sql
                    , connection);
                commander.Parameters.AddWithValue("@numeEchipa", entity.NumeEchipa);
                commander.Prepare();

                commander.ExecuteNonQuery();
                
                Console.WriteLine("Row Inserted");
                connection.Close();

            }
            catch (Exception e)
            {
                Console.WriteLine(e);
                throw;
            }
        }

        public void Update(int id,Echipa echipa)
        {
            try
            {
                _validator.validate(echipa);
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
                return;
            }
            throw new System.NotImplementedException();
        }

        public void Delete(int id)
        {
            throw new System.NotImplementedException();
        }

        public MySqlDataAdapter getDataAdapter()
        {
            
            dataAdapter.SelectCommand = new MySqlCommand("select * from echipa", _utils.GetConnection());
            return dataAdapter;
        }
    }
}

File: C#JAVA_Client_Server/ClientServcerC#Protocols/LongExample/Repo/src/main/java/repository/database/EchipaDBRepository.java
package repository.database;


import entities.domain.Echipa;
import entities.domain.validator.interfaceAndExceptions.Validator;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import repository.EchipaRepository;
import repository.JdbcUtils;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class EchipaDBRepository implements EchipaRepository {
    private Validator<Echipa> validator;

    private JdbcUtils dbUtils;
    private static final Logger logger= LogManager.getLogger(EchipaDBRepository.class);

    public EchipaDBRepository(Properties props, Validator<Echipa> validator) {
        this.validator=validator;
        logger.info("Initializing CarsDBRepository with properties: {} ",props);
        this.dbUtils =new JdbcUtils(props);    }

    @Override
    public Echipa findOne(Integer integer) {
        logger.traceEntry();
        Connection con= dbUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("select * from echipa where idEchipa=?"))
        {
            preStmt.setInt(1,integer);
            try(ResultSet resultSet=preStmt.executeQuery()){
                resultSet.next();
                int id=resultSet.getInt("idEchipa");
                String numeEchipa=resultSet.getString("numeEchipa");
                Echipa echipa=new Echipa(numeEchipa);
                echipa.setId(id);
                return echipa;
            }
        }catch (SQLException e){
            logger.error(e);
            System.err.println("Error DB"+e);
        }
        try {
            con.close();
        } catch (SQLException throwables) {
            logger.error(throwables);
            System.err.println("ERROR DB"+throwables);
        }
        logger.traceExit(integer);
        return null;
    }

    @Override
    public Iterable<Echipa> findAll() {
        logger.traceEntry();
        Connection con= dbUtils.getConnection();
        List<Echipa> echipe=new ArrayList<>();
        try(PreparedStatement preStmt=con.prepareStatement("select * from echipa"))
        {
            try(ResultSet resultSet=preStmt.executeQuery()){
                while (resultSet.next()){
                    int id=resultSet.getInt("idEchipa");
                    String numeEchipa=resultSet.getString("numeEchipa");
                    Echipa echipa=new Echipa(numeEchipa);
                    echipa.setId(id);
                    echipe.add(echipa);
                }
            }
        }catch (SQLException e){
            logger.error(e);
            System.err.println("Error DB"+e);
        }
        try {
            con.close();
        } catch (SQLException throwables) {
            logger.error(throwables);
            System.err.println("ERROR DB"+throwables);
        }
        logger.traceExit(echipe);
        return echipe;
    }

    @Override
    public void save(Echipa echipa) {
        logger.traceEntry("saving task {}",echipa);
        try{
            validator.validate(echipa);
        }catch (Exception e){
            logger.error(e);
            System.err.println("ERROR DOMAIN"+e);
            logger.traceExit();
            return;
        }
        Connection con= dbUtils.getConnection();
        try (PreparedStatement preparedStatement=con.prepareStatement("insert into echipa (numeEchipa)values (?)")){
            preparedStatement.setString(1,echipa.getNumeEchipa());
            int result=preparedStatement.executeUpdate();
            logger.trace("Saved {} instance",result);
        } catch (SQLException throwables) {
            logger.error(throwables);
            System.err.println("ERROR DB"+throwables);
        }
        try {
            con.close();
        } catch (SQLException throwables) {
            logger.error(throwables);
            System.err.println("ERROR DB"+throwables);
        }
        logger.traceExit();
    }

    @Override
    public void update(Integer integer,Echipa echipa) {
        logger.traceEntry("update task {}",integer);
        try{
            validator.validate(echipa);
        }catch (Exception e){
            logger.error(e);
            System.err.println("ERROR DOMAIN"+e);
            logger.traceExit();
            return;
        }
        Connection con= dbUtils.getConnection();
        try (PreparedStatement preparedStatement=con.prepareStatement("update echipa set numeEchipa=? where idEchipa=?")){
            preparedStatement.setInt(1,integer);
            preparedStatement.setString(2,"Merge");
            int result=preparedStatement.executeUpdate();
            logger.trace("Updated {} instance",result);
        } catch (SQLException throwables) {
            logger.error(throwables);
            System.err.println("ERROR DB"+throwables);
        }
        try {
            con.close();
        } catch (SQLException throwables) {
            logger.error(throwables);
            System.err.println("ERROR DB"+throwables);
        }
        logger.traceExit();
    }

    @Override
    public void delete(Integer integer) {
        logger.traceEntry("delete task {}",integer);
        Connection con= dbUtils.getConnection();
        try (PreparedStatement preparedStatement=con.prepareStatement("delete from echipa where idEchipa=?")){
            preparedStatement.setInt(1,integer);
            int result=preparedStatement.executeUpdate();
            logger.trace("deleted {} instance",result);
        } catch (SQLException throwables) {
            logger.error(throwables);
            System.err.println("ERROR DB"+throwables);
        }
        try {
            con.close();
        } catch (SQLException throwables) {
            logger.error(throwables);
            System.err.println("ERROR DB"+throwables);
        }
        logger.traceExit();
    }
}
